components/Elevation/src/UIView+MaterialElevationResponding.m

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "UIView+MaterialElevationResponding.h"
+
+#import "MDCElevation.h"
+#import "MDCElevationOverride.h"
+
+@implementation UIView (MaterialElevationResponding)
+
+- (CGFloat)mdc_baseElevation {
+  CGFloat totalElevation = 0;
+  UIView *current = self;
+
+  while (current != nil) {
+    id<MDCElevation> elevatableCurrent = [current objectConformingToElevationInResponderChain];
```

Comment:
Just curious, it seems the assumption here is only one of UIView and its UIViewController can support elevation/can conform to the protocol. Is this something worth documented?

Reply:
Added docs around UIViewControllers behavior. I think we safe not saying that it doesn't support things that are not UIViewControllers or UIViews since it is a category on UIView, what do you think?

Reply:
I think that's fine. It is kinda an edge case when I was reading the code, so I was just trying to make it clear. Thanks for the clarification:)